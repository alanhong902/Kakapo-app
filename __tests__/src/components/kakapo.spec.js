import React from 'react';
import { identity, values } from 'ramda';
import { ImportKakapo } from 'components/import/kakapo';
import { getData, createComponentWithIntl, randomSounds } from '../helper';

function setup(props = {}) {
  const propData = {
    ...getData('search'),
    soundActions: {},
    dispatch: identity,
    ...props,
  };
  return {
    props,
    tree: createComponentWithIntl(<ImportKakapo {...propData} />).toJSON(),
  };
}

test('<Kakapo/> render', () => {
  const { tree } = setup();
  expect(tree).toMatchSnapshot();
});

test('<Kakapo/> render items', () => {
  const { tree } = setup({
    search: { kakapofavs: values(randomSounds(5)) },
  });
  expect(tree).toMatchSnapshot();
});
