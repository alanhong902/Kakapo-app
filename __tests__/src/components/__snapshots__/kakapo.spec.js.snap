// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Kakapo/> render 1`] = `
<div
  className="kakapo"
>
  <h5>
    <span>
      Add a sound from Kakapo.com
    </span>
  </h5>
  <div
    className=""
  />
</div>
`;

exports[`<Kakapo/> render items 1`] = `
<div
  className="kakapo"
>
  <h5>
    <span>
      Add a sound from Kakapo.com
    </span>
  </h5>
  <div
    className="kakapofavs-items"
  >
    <div
      className="kakapo-item"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <div
        className="thumbnail"
      >
        <i
          className="icon-null"
        />
      </div>
      <span
        className="title"
      >
        <span>
          Thunder
        </span>
      </span>
    </div>
    <div
      className="kakapo-item"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <div
        className="thumbnail"
      >
        <i
          className="icon-null"
        />
      </div>
      <span
        className="title"
      >
        <span>
          Thunder
        </span>
      </span>
    </div>
    <div
      className="kakapo-item"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <div
        className="thumbnail"
      >
        <i
          className="icon-null"
        />
      </div>
      <span
        className="title"
      >
        <span>
          Thunder
        </span>
      </span>
    </div>
    <div
      className="kakapo-item"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <div
        className="thumbnail"
      >
        <i
          className="icon-null"
        />
      </div>
      <span
        className="title"
      >
        <span>
          Thunder
        </span>
      </span>
    </div>
    <div
      className="kakapo-item"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <div
        className="thumbnail"
      >
        <i
          className="icon-null"
        />
      </div>
      <span
        className="title"
      >
        <span>
          Thunder
        </span>
      </span>
    </div>
  </div>
</div>
`;
